core/templates/dev/head/dashboard/Dashboard.js

```diff
@@ -50,25 +61,55 @@ oppia.controller('Dashboard', [
       $window.location = '/create/' + explorationId;
     };
 
+    $scope.openExploration = function(status, explorationId) {
+      if (status === 'private') {
+        $scope.showExplorationEditor(explorationId);
+      } else {
+        var isOpen = !$scope.isDropdownOpen[explorationId];
+        $scope.isDropdownOpen = {};
+        $scope.isDropdownOpen[explorationId] = isOpen;
+      }
+    };
+
     $scope.myExplorationsView = 'card';
     $scope.setMyExplorationsView = function(viewType) {
       $scope.myExplorationsView = viewType;
     };
 
+    $scope.isMobileView = function() {
+      return ($window.innerWidth < 500);
+    };
+
     $scope.checkForMobileView = function() {
-      if ($window.innerWidth < 500) {
+      if ($scope.isMobileView()) {
         $scope.myExplorationsView = 'card';
         $scope.publishText = EXP_PUBLISH_TEXTS.smText;
       } else {
         $scope.publishText = EXP_PUBLISH_TEXTS.defaultText;
       }
     };
 
+    $scope.isDropdownOpenOnMobile = function(explorationId) {
+      if ($scope.isMobileView() && $scope.isDropdownOpen[explorationId]) {
+        return true;
+      }
+      return false;
+    };
+
     $scope.checkForMobileView();
     angular.element($window).bind('resize', function() {
       $scope.checkForMobileView();
     });
 
+    $scope.toggleDropdown = function(type, explorationId, event) {
```

Comment:
Hard to understand what this function does -- can you add docs about what 'type' represents?